src/main/java/sprouts/impl/AbstractVariables.java

```diff
@@ -129,82 +128,110 @@ protected AbstractVariables( boolean isImmutable, Class<T> type, boolean allowsN
     @Override public final int size() { return _variables.size(); }
 
     /** {@inheritDoc} */
-    @Override public Vars<T> removeLast( int count )
-    {
-        if ( _isImmutable ) throw new UnsupportedOperationException( "This list is immutable." );
+    @Override
+    public Vars<T> removeLast( int count ) {
+        if ( _isImmutable )
+            throw new UnsupportedOperationException( "This list is immutable." );
+        if ( count < 0)
+            throw new IllegalArgumentException("Invalid count! Count must be non-negative.");
+
         count = Math.min( count, size() );
-        if ( count == 0 ) return this;
-        if ( count == 1 ) return removeLast();
-        for ( int i = 0; i < count; i++ ) _variables.remove( size() - 1 );
-        _triggerAction( Change.REMOVE, -1, null, null );
+
+        if ( count == 0 )
+            return this;
+        if ( count == 1 )
+            return removeLast();
+
+        Vars<T> vars = (Vars<T>) (_allowsNull ? Vars.ofNullable(_type) : Vars.of(_type));
+
+        List<Var<T>> subList = _variables.subList( size() - count, size() );
+        for ( Var<T> var : subList ) vars.add(var);
+        subList.clear();
+
+        _triggerAction( Change.REMOVE, size(), null, vars );
```

Comment:
should it not be `_triggerAction( Change.REMOVE, size() - count, null, vars );`? 
Since that is the index at which the removal
of elements starts...